File: Practica 1/S2/CryptoVisionaryTUI/src/FactoriaAbstracta/CryptoCurrency.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package FactoriaAbstracta;


public class CryptoCurrency {
    protected String token;
    protected Double valorActual;
    protected Double valorMinimo;
    protected Double valorMaximo;
    
    
    
    public void setToken(){
        
    }
    
    public void setValorActual(double nuevoValor){
        
         if(nuevoValor < this.valorMinimo ){
               this.setValorMinimo(nuevoValor);
            }
            
            if(nuevoValor > valorMaximo){
                this.setValorMaximo(nuevoValor);
            }
      this.valorActual=nuevoValor; 
    }
    
    public void setValorMaximo(double nuevoValor){
        this.valorMaximo=nuevoValor;
    }
    
    public void setValorMinimo(double nuevoValor){
        this.valorMinimo=nuevoValor;
    }

    public String getToken() {
        return token;
    }

    public Double getValorActual() {
        return valorActual;
    }

    public Double getValorMinimo() {
        return valorMinimo;
    }

    public Double getValorMaximo() {
        return valorMaximo;
    }
    
    
    
    public void run(){
        
    }
    
    
}

File: Practica 1/S3/CryptoFilters/CryptoCurrency.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   CrytpoCurrency.cpp
 * Author: reko98
 * 
 * Created on 25 de marzo de 2021, 12:55
 */

#include "CryptoCurrency.h"

CryptoCurrency::CryptoCurrency() {
}

CryptoCurrency::CryptoCurrency(const CryptoCurrency& orig) {
}

CryptoCurrency::~CryptoCurrency() {
}

void CryptoCurrency::execution(){
	cout << "El valor Final de la CriptoMoneda despues de aplicar los filtros es: " 
	     << this->getValorActual() << endl;
	
}
